### PR TITLE
Use parameters for VPC and subnets CIDR blocks

### LINKED ISSUE #183: Ticket 148: Choose Subnets for new VPC template
Would be great to be able to choose the subnet when creating a cluster with a new vpc. This would allow more folks to peer the cluster since they could be sure it didn't overlap with any existing vpc.

### PR DESCRIPTION
**Pull request by vincepri**
Fixes #183

### CODE REVIEW

File: templates/kubernetes-cluster-with-new-vpc.template [hunk 1]
```diff
@@ -75,6 +84,27 @@ Metadata:
...
 Parameters:
+  VPCCIDR:
+    Description: VPC CIDR Block
+    Type: String
+    AllowedPattern: "(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})/(\\d{1,2})"
+    ConstraintDescription: must be a valid IP CIDR range of the form x.x.x.x/x.
+    Default: '10.0.0.0/16'
+
+  PrivateSubnetCIDR:
+    Description: CIDR Block for the Private Subnet
```
Comment (detiber) [inline templates/kubernetes-cluster-with-new-vpc.template]: The description should probably mention that this must be a valid subnet of the VPCCIDR and not conflict with PublicSubnetCIDR.

File: templates/kubernetes-cluster-with-new-vpc.template [hunk 2]
```diff
@@ -75,6 +84,27 @@ Metadata:
...
+    Type: String
+    AllowedPattern: "(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})/(\\d{1,2})"
+    ConstraintDescription: must be a valid IP CIDR range of the form x.x.x.x/x.
+    Default: '10.0.0.0/19'
+
+  PublicSubnetCIDR:
+    Description: CIDR Block for the Public Subnet
```
Comment (detiber) [inline templates/kubernetes-cluster-with-new-vpc.template]: The description should probably mention that this must be a valid subnet of the VPCCIDR and not conflict with PrivateSubnetCIDR.

**Comment by detiber** [comment]
@rbankston ptal

**Comment by rbankston** [comment]
LGTM thanks for the turnaround

**Pull request merged**